### PR TITLE
More thorough testing against ASE

### PR DESCRIPTION
**Pull request by frostedoyster**
This PR adds a few systems (more details in `python/vesin/tests/data/readme.txt` and extends the ASE tests to more potential input combinations

### CODE REVIEW

File: python/vesin/tests/data/readme.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+This is a small test suite for vesin, composed of:
+- water.xyz: a medium-sized cluster
+- napthalene.xyz: a small organic molecule
+- diamond.xyz: a periodic system with a cubic cell
+- carbon.xyz: a periodic system with a non-cubic cell
+
+TODO: add systems with boundary conditions along some directions, but not others
```
Comment (Luthaf) [inline python/vesin/tests/data/readme.txt:7]: this is not implemented

File: python/vesin/tests/test_neighbors.py [hunk 2]
```diff
@@ -1,28 +1,57 @@
...
 
 import vesin
 
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+
 
-def test_neighbors():
-    atoms = ase.build.make_supercell(ase.build.bulk("Si", "diamond"), 3 * np.eye(3))
+@pytest.mark.parametrize("system", ["water", "diamond", "naphthalene", "carbon"])
+@pytest.mark.parametrize("quantities", ["ijSD", "ijS"])
```
Comment (Luthaf) [inline python/vesin/tests/test_neighbors.py:15]: I don't see the point of this parametrization, the second one just computes a subset of the first one.

Comment (frostedoyster) [reply python/vesin/tests/test_neighbors.py:15]: Hmm well, given that I didn't check how the library works, I thought it was a good idea. But if you think ijS will always be correct if ijSD is, then it can be removed

Comment (Luthaf) [reply python/vesin/tests/test_neighbors.py:15]: yes, everything is done at the same time, it's only a question of do we throw away this data or do we store it in some allocation
